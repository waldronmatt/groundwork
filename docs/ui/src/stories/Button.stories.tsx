import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button, type ButtonProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta = {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
  parameters: {
    badges: ['stable'],
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
} satisfies Meta<typeof Button>;

export default meta;

const getCaptionForLocale = (locale: string) => {
  switch (locale) {
    case 'es':
      return 'Hola!';
    default:
      return 'Hello!';
  }
};

const onClickHandler = action('button-click');

const Template: StoryFn<typeof Button> = (args: ButtonProps, { globals: { locale } }) => {
  // by default, let the user modify the text through the `children` control
  // otherwise, load in the text from locale for i18n testing
  const caption = locale !== '' ? getCaptionForLocale(locale as string) : args.children;
  return (
    <Button {...args} onClick={onClickHandler}>
      {caption}
    </Button>
  );
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});
export const Hover = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};

Hover.args = {
  ...defaultProps,
  variant: 'md',
};

Hover.parameters = {
  pseudo: { hover: true },
};
