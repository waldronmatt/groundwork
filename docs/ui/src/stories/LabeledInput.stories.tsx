import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { LabeledInput, type LabeledInputProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof LabeledInput> = {
  title: 'Components/LabeledInput',
  component: LabeledInput,
  argTypes: {
    variant: {
      table: {
        disable: true,
      },
    },
  },
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
};

export default meta;

const onChangeHandler = action('labeledInput-change');
const onFocusHandler = action('labeledInput-focus');

const Template: StoryFn<typeof LabeledInput> = (args: LabeledInputProps) => {
  return <LabeledInput {...args} onChange={onChangeHandler} onFocus={onFocusHandler} />;
};

export const Default = Template.bind({});

Default.args = {
  ...defaultProps,
  id: 'unique_id',
};
