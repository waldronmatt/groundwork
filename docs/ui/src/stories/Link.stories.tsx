import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Link, type LinkProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
  href: '/',
};

const meta = {
  title: 'Components/Link',
  component: Link,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
  parameters: {
    badges: ['stable'],
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
} satisfies Meta<typeof Link>;

export default meta;

const getCaptionForLocale = (locale: string) => {
  switch (locale) {
    case 'es':
      return 'Hola!';
    default:
      return 'Hello!';
  }
};

const onClickHandler = action('link-click');

const Template: StoryFn<typeof Link> = (args: LinkProps, { globals: { locale } }) => {
  // by default, let the user modify the text through the `children` control
  // otherwise, load in the text from locale for i18n testing
  const caption = locale !== '' ? getCaptionForLocale(locale as string) : args.children;
  return (
    <Link
      {...args}
      onClick={(event) => {
        event.preventDefault();
        onClickHandler(event);
      }}
    >
      {caption}
    </Link>
  );
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};
