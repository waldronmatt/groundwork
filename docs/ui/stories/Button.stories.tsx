import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button, type ButtonProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const onClickHandler = action('button-click');

const Template: StoryFn<typeof Button> = (args: ButtonProps) => {
  return <Button {...args} onClick={onClickHandler} />;
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};
