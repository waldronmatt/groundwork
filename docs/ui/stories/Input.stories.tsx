import type { Meta, StoryFn } from '@storybook/react';
// we use `lib/` paths so we can get source files and have storybook
// auto refresh (hmr) whenever we update our component source files
import { Input, type InputProps } from '@waldronmatt/demo-ui/lib/index.js';

const meta: Meta<typeof Input> = {
  title: 'Components/Input',
  component: Input,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const Template: StoryFn<typeof Input> = (args: InputProps) => {
  return <Input {...args}></Input>;
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  variant: 'md',
};

Small.args = {
  variant: 'sm',
};

Large.args = {
  variant: 'lg',
};
