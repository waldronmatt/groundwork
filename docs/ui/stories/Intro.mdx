# Demo UI Components

Welcome to the Demo UI Component Library. See the [component library readme](https://github.com/waldronmatt/pnpm-nx-lerna-lite-boilerplate/blob/main/packages/ui/README.md) for more information.
