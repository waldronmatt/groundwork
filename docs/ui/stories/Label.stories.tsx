import type { Meta, StoryFn } from '@storybook/react';
// we use `lib/` paths so we can get source files and have storybook
// auto refresh (hmr) whenever we update our component source files
import { Label, type LabelProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof Label> = {
  title: 'Components/Label',
  component: Label,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const Template: StoryFn<typeof Label> = (args: LabelProps) => {
  return <Label {...args}></Label>;
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};
