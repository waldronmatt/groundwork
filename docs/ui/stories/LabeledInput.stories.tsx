import type { Meta, StoryFn } from '@storybook/react';
// we use `lib/` paths so we can get source files and have storybook
// auto refresh (hmr) whenever we update our component source files
import { LabeledInput, type LabeledInputProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof LabeledInput> = {
  title: 'Components/LabeledInput',
  component: LabeledInput,
};

export default meta;

const Template: StoryFn<typeof LabeledInput> = (args: LabeledInputProps) => {
  return <LabeledInput {...args}></LabeledInput>;
};

export const Default = Template.bind({});

Default.args = {
  ...defaultProps,
  id: 'unique_id',
};
