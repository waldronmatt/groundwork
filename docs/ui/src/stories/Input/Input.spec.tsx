import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { Input } from '@waldronmatt/demo-ui/lib/index.js';
import { describe, it, expect, vi } from 'vitest';
import { composeStories } from '@storybook/react';
import * as stories from './Input.stories.js';

const { Default, Small, Large } = composeStories(stories);

describe('Input component', () => {
  it('renders with default props', () => {
    const { container } = render(<Default />);
    const inputElement = screen.getByRole('textbox');

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveClass('input');
    expect(inputElement).toHaveClass('md');
    expect(container).toMatchSnapshot();
  });

  it('renders with sm variant', () => {
    const { container } = render(<Small />);
    const inputElement = screen.getByRole('textbox');

    expect(inputElement).toHaveClass('sm');
    expect(container).toMatchSnapshot();
  });

  it('renders with lg variant', () => {
    const { container } = render(<Large variant="lg" />);
    const inputElement = screen.getByRole('textbox');

    expect(inputElement).toHaveClass('lg');
    expect(container).toMatchSnapshot();
  });

  it('handles native props', () => {
    const { container } = render(<Input id="testId" />);
    const inputElement = screen.getByRole('textbox');

    expect(inputElement).toHaveAttribute('id', 'testId');
    expect(container).toMatchSnapshot();
  });

  it('handles input events', () => {
    const onChangeMock = vi.fn();
    render(<Input onChange={onChangeMock} />);
    const inputElement = screen.getByRole('textbox');
    const eventTargetObj = { target: { value: 'Hello World!' } };
    fireEvent.change(inputElement, eventTargetObj);

    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });
});
