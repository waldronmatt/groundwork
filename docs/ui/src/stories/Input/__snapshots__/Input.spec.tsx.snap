// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > handles native props 1`] = `
<div>
  <input
    class="input md"
    id="testId"
  />
</div>
`;

exports[`Input component > renders with default props 1`] = `
<div>
  <input
    class="input md"
  />
</div>
`;

exports[`Input component > renders with lg variant 1`] = `
<div>
  <input
    class="input lg"
  />
</div>
`;

exports[`Input component > renders with sm variant 1`] = `
<div>
  <input
    class="input sm"
  />
</div>
`;
